src/main/java/com/devcycle/sdk/server/openfeature/DevCycleProvider.java

```diff
@@ -0,0 +1,144 @@
+package com.devcycle.sdk.server.openfeature;
+
+import com.devcycle.sdk.server.common.api.IDevCycleClient;
+import com.devcycle.sdk.server.common.model.DevCycleUser;
+import com.devcycle.sdk.server.common.model.Variable;
+import dev.openfeature.sdk.*;
+import dev.openfeature.sdk.exceptions.ProviderNotReadyError;
+import dev.openfeature.sdk.exceptions.TypeMismatchError;
+
+import java.util.Map;
+
+public class DevCycleProvider implements FeatureProvider {
+    private static final String PROVIDER_NAME = "DevCycleProvider";
+
+    private final IDevCycleClient devcycleClient;
+
+    public DevCycleProvider(IDevCycleClient devcycleClient) {
+        this.devcycleClient = devcycleClient;
+    }
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> PROVIDER_NAME + " " + devcycleClient.getSDKPlatform();
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue, EvaluationContext ctx) {
+        return resolvePrimitiveVariable(key, defaultValue, ctx);
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(String key, String defaultValue, EvaluationContext ctx) {
+        return resolvePrimitiveVariable(key, defaultValue, ctx);
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue, EvaluationContext ctx) {
+        ProviderEvaluation<Integer> eval = resolvePrimitiveVariable(key, defaultValue, ctx);
+        return eval;
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue, EvaluationContext ctx) {
+        return resolvePrimitiveVariable(key, defaultValue, ctx);
+    }
+
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(String key, Value defaultValue, EvaluationContext ctx) {
+        /*
+         * JSON objects have special rules in the DevCycle SDK and must be handled differently
+         * - must always be an object, no lists or literal values
+         * - must only contain strings, numbers, and booleans
+         */
+        if (!defaultValue.isStructure()) {
+            throw new TypeMismatchError("Default value must be a OpenFeature structure");
+        }
+
+        for (String k : defaultValue.asStructure().keySet()) {
+            Value v = defaultValue.asStructure().getValue(k);
+            if (!(v.isString() || v.isNumber() || v.isBoolean() || v.isNull())) {
+                throw new TypeMismatchError("DevCycle JSON objects may only contain strings, numbers, booleans and nulls");
+            }
+        }
```

Comment:
Why do we need to check the keys of the default object as well? I didn't do this on the Go side, just wondering if I missed something.

Reply:
I added this after I saw Jamie do it on the C# SDK but also the fact that we call out these restrictions in the docs so I figured we should enforce them.  In Java the keys can't be null and the map is forced to use Strings for them.